iceoryx_utils/cmake/IceoryxPlatform.cmake

```diff
@@ -33,10 +33,13 @@ endif(LINUX)
 
 if(CMAKE_CXX_COMPILER_ID MATCHES "MSVC")
     set(ICEORYX_WARNINGS PRIVATE ${ICEORYX_WARNINGS} /W1)
+    # todo: '/O2' and '/RTC1' (set by default) options are incompatible, 
 elseif(CMAKE_CXX_COMPILER_ID MATCHES "GNU")
     set(ICEORYX_WARNINGS PRIVATE ${ICEORYX_WARNINGS} -W -Wall -Wextra -Wuninitialized -Wpedantic -Wstrict-aliasing -Wcast-align -Wno-noexcept-type -Wconversion)
+    set(OPTIMIZE_OPTIONS -O2)
```

Comment:
@all Is the O2 optimization agreed by all?

Reply:
@budrus @mossmaurice @MatthiasKillat @FerdinandSpitzschnueffler @elBoberido see question of @dkroenke

Reply:
no, that's what `-DCMAKE_BUILD_TYPE` is for. This change doesn't differentiate between release and debug builds and makes debugging harder. Please use just `-DCMAKE_BUILD_TYPE` for building with optimizations.
Debug has `-g`
Release has `-O3 -DNDEBUG`
RelWithDebInfo `-O2 -DNDEBUG`
MinSizerel has `-Os -DNDEBUG`
This are from cmake 3.16